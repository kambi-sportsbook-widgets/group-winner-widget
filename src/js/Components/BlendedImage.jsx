import React from 'react'
import PropTypes from 'prop-types'
import styles from './BlendedImage.scss'

const BlendedImage = ({ imgPath }) => (
  <svg
    className={styles.general}
    xmlnsXlink="http://www.w3.org/1999/xlink&quot; xmlns=&quot;http://www.w3.org/2000/svg"
  >
    <defs>
      <mask id="svgmask2">
        <image width="100%" height="100%" xlinkHref={imgPath} />
      </mask>
    </defs>
    <rect
      mask="url(#svgmask2)"
      className={`KambiWidget-primary-color ${styles.blendRect}`}
      x="0"
      y="0"
      width="100%"
      height="100%"
    />
  </svg>
)

BlendedImage.propTypes = {
  /**
   * Image path
   */
  imgPath: PropTypes.string.isRequired,
}

export default BlendedImage
