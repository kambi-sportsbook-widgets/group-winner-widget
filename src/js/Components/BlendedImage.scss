:local(.general) {
   width: 100%;
   height: 100%;
}
