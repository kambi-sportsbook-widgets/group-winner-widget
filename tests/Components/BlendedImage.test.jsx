/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import BlendedImage from '../../src/js/Components/BlendedImage';

let renderer;

describe('BlendedImage.render()', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <BlendedImage imgPath="test/image/path.jpg" />
      )).toMatchSnapshot();
   });

});
