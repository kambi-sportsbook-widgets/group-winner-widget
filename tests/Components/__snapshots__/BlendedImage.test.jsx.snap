// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlendedImage.render() renders correctly with default props 1`] = `
<svg
  className="general"
  xmlnsXlink="http://www.w3.org/1999/xlink\\" xmlns=\\"http://www.w3.org/2000/svg"
>
  <defs>
    <mask
      id="svgmask2"
    >
      <image
        height="100%"
        width="100%"
        xlinkHref="test/image/path.jpg"
      />
    </mask>
  </defs>
  <rect
    height="100%"
    id="blendRect"
    mask="url(#svgmask2)"
    width="100%"
    x="0"
    y="0"
  />
</svg>
`;
